sub-packages/bionemo-geometric/src/bionemo/geometric/atom_featurizers.py

```diff
@@ -0,0 +1,400 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: LicenseRef-Apache2
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from pathlib import Path
+from typing import List
+
+import numpy as np
+import pandas as pd
+from rdkit import Chem
+from rdkit.Chem import Atom, rdMolDescriptors
+from rdkit.Chem.rdchem import ChiralType, HybridizationType
+from rdkit.Chem.Scaffolds import MurckoScaffold
+
+from bionemo.geometric.base_featurizer import (
+    BaseFeaturizer,
+    get_boolean_atomic_prop,
+    get_double_atomic_prop,
+    one_hot_enc,
+)
+
+
+# Extremum for molar refractivity
+MIN_MR = 0.0
+MAX_MR = 6.0
+
+# Extremum for logP
+MIN_LOGP = -2.996
+MAX_LOGP = 0.8857
+
+ALL_ATOM_FEATURIZERS = [
+    "PeriodicTableFeaturizer",
+    "ElectronicPropertyFeaturizer",
+    "ScaffoldFeaturizer",
+    "SmartsFeaturizer",
+    "CrippenFeaturizer",
+]
+
+MAX_ATOMIC_NUM = 100
+MAX_CHIRAL_TYPES = len(ChiralType.values)
+MAX_HYBRIDIZATION_TYPES = len(HybridizationType.values)
+MAX_NUM_HS = 5  # 4 + 1 (no hydrogens)
+
+
+class AtomicNumberFeaturizer(BaseFeaturizer):
+    """Class for featurizing atom by its atomic number."""
+
+    def __init__(self) -> None:
+        """Initializes AtomicNumberFeaturizer class."""
+        pass
+
+    @property
+    def n_dim(self) -> int:
+        """Returns dimensionality of the computed features."""
+        return MAX_ATOMIC_NUM
+
+    def get_features(self, atom: Atom) -> List[bool]:
+        """Returns features of the atom."""
+        return one_hot_enc(atom.GetAtomicNum() - 1, MAX_ATOMIC_NUM)
+
+
+class DegreeFeaturizer(BaseFeaturizer):
+    """Class for featurizing atom by its degree (excluding hydrogens) of connectivity."""
+
+    def __init__(self) -> None:
+        """Initializes DegreeFeaturizer class."""
+        pass
+
+    @property
+    def n_dim(self) -> int:
+        """Returns dimensionality of the computed features."""
+        return 6
+
+    def get_features(self, atom: Atom) -> List[bool]:
+        """Returns features of the atom."""
+        return one_hot_enc(atom.GetDegree(), self.n_dim)
```

Comment:
Here, and elsewhere, might make sense to separate the one-hot transformation from the feature itself.

Something like
```python
OneHotDegreeFeaturizer = OneHot(DegreeFeaturizer)
```
that you could re-use in a bunch of places